In [1]:
import numpy as np
import pandas as pd
from scipy import stats

# EXERCISE 1. The hourly wages in a particular industry are normally distributed with mean $13.20 and standard deviation $2.50. A company in this industry employs 40 workers, paying them an average of $12.20 per hour. Can this company be accused of paying substandard wages? Use an α = .01 level test.

In [2]:
mean1 = 13.20
mean2 = 12.20
std = 2.50
sample = 40

In [3]:
z_test = (mean2 - mean1) / (std / np.sqrt(sample))
z_test

-2.5298221281347035

-z0.01 = -2.326

-2.53 < -z0.01. So, H0 is rejected: there is evidence that the company is paying substandard wages.

# EXERCISE 2. Shear strength measurements derived from unconfined compression tests for two types of soils gave the results shown in the following document (measurements in tons per square foot). Do the soils appear to differ with respect to average shear strength, at the 1% significance level?

In [4]:
soil = pd.read_csv("/content/soil - Sheet1.csv")
soil.head()

Soil1  Soil2
0  1.442  1.364
1  1.943  1.878
2  1.110  1.337
3  1.912  1.828
4  1.553  1.371

In [5]:
stats.ttest_ind(soil["Soil1"], soil["Soil2"], nan_policy="omit")

Ttest_indResult(statistic=5.1681473319343345, pvalue=2.5932287323528147e-06)

With α = .01, the test rejects if |t| > 2.576. So, we can reject the hypothesis that the soils have equal mean shear strengths.

# EXERCISE 3. The following dataset is based on data provided by the World Bank (https://datacatalog.worldbank.org/dataset/education-statistics). World Bank Edstats.

1. Get descriptive statistics (the central tendency, dispersion and shape of a dataset’s distribution) for each continent group (AS, EU, AF, NA, SA, OC).

In [6]:
import numpy as np
import pandas as pd
from scipy import stats

In [7]:
df = pd.read_csv("/content/2015 PISA Test - Sheet1.csv")
df.head()

Country Code Continent_Code  ...   Reading   Science
0          ALB             EU  ...  405.2588  427.2250
1          ARE             AS  ...  433.5423  436.7311
2          ARG             SA  ...  425.3031  432.2262
3          AUS             OC  ...  502.9006  509.9939
4          AUT             EU  ...  484.8656  495.0375

[5 rows x 6 columns]

In [8]:
df.groupby(["Continent_Code"], dropna=False).describe()

internet_users_per_100             ...     Science          
                                count       mean  ...         75%       max
Continent_Code                                    ...                      
AF                                2.0  43.359918  ...  383.738825  386.4034
AS                               17.0  68.455613  ...  523.277400  555.5747
EU                               37.0  77.274888  ...  501.936900  534.1937
OC                                2.0  86.391704  ...  512.476100  513.3035
SA                                7.0  60.180494  ...  433.794600  446.9561
NaN                               5.0  66.886792  ...  496.242400  527.7047

[6 rows x 32 columns]

In [55]:
df_mean = df.groupby(["Continent_Code"], dropna=False).mean()
df_mean

internet_users_per_100        Math     Reading     Science
Continent_Code                                                            
AF                           43.359918  363.212100  355.457400  381.074250
AS                           68.455613  466.216647  454.059682  467.945847
EU                           77.274888  477.981449  476.164608  478.299381
OC                           86.391704  494.559750  506.085650  511.648700
SA                           60.180494  402.887700  425.359229  421.747186
NaN                          66.886792  424.250960  446.420920  438.180760

In [56]:
df_median = df.groupby(["Continent_Code"], dropna=False).median()
df_median

internet_users_per_100       Math    Reading    Science
Continent_Code                                                         
AF                           43.359918  363.21210  355.45740  381.07425
AS                           74.000000  459.81600  433.54230  456.48360
EU                           76.184000  489.72870  487.25010  490.22500
OC                           86.391704  494.55975  506.08565  511.64870
SA                           64.289000  409.03330  425.30310  424.59050
NaN                          59.762950  408.02350  427.48750  419.60800

In [57]:
df_mode = df.groupby(["Continent_Code"], dropna=False).agg(lambda x:x.value_counts().index[0])
df_mode

Country Code  internet_users_per_100  ...   Reading   Science
Continent_Code                                       ...                    
AF                      TUN               38.200000  ...  361.0555  375.7451
AS                      KAZ               21.976068  ...  535.1002  524.6445
EU                      LVA               71.378000  ...  416.2293  464.7819
OC                      AUS               84.560519  ...  509.2707  513.3035
SA                      TTO               40.900000  ...  407.3486  446.9561
NaN                     DOM               59.762950  ...  423.2765  527.7047

[6 rows x 5 columns]

In [58]:
df_std = df.groupby(["Continent_Code"], dropna=False).std()
df_std

internet_users_per_100       Math    Reading    Science
Continent_Code                                                         
AF                            7.297226   5.099513   7.916909   7.536556
AS                           21.086060  64.356490  57.048962  56.671371
EU                           12.425773  35.150403  37.720545  34.450616
OC                            2.589686   0.938401   4.504341   2.340241
SA                            9.772455  18.128894  19.709688  18.470319
NaN                          14.364359  71.717058  66.614744  76.789918

In [59]:
df_variance = df.groupby(["Continent_Code"], dropna=False).var()
df_variance

internet_users_per_100         Math      Reading      Science
Continent_Code                                                               
AF                           53.249507    26.005030    62.677447    56.799679
AS                          444.621918  4141.757822  3254.584049  3211.644321
EU                          154.399841  1235.550805  1422.839540  1186.844969
OC                            6.706475     0.880597    20.289087     5.476726
SA                           95.500869   328.656783   388.471816   341.152678
NaN                         206.334814  5143.336351  4437.524053  5896.691519

In [14]:
Q1 = df.groupby(["Continent_Code"], dropna=False).quantile([0.25])
Q1

internet_users_per_100        Math     Reading     Science
Continent_Code                                                                 
AF             0.25               40.779959  361.409150  352.658350  378.409675
AS             0.25               50.300000  403.833200  408.102200  417.611200
EU             0.25               68.632900  464.040100  452.514300  460.774900
OC             0.25               85.476112  494.227975  504.493125  510.821300
SA             0.25               57.116462  388.102200  416.126900  408.205450
NaN            0.25               57.431043  400.253400  423.276500  415.709900

In [15]:
Q3 = df.groupby(["Continent_Code"], dropna=False).quantile([0.75])
Q3

internet_users_per_100        Math     Reading     Science
Continent_Code                                                                 
AF             0.75               45.939877  365.015050  358.256450  383.738825
AS             0.75               84.948353  531.296100  508.690500  523.277400
EU             0.75               87.479056  503.722000  499.814600  501.936900
OC             0.75               87.307296  494.891525  507.678175  512.476100
SA             0.75               66.321532  417.617650  431.922700  433.794600
NaN            0.75               74.554202  469.628500  496.935100  496.242400

In [16]:
grouper = df.groupby("Continent_Code", dropna=False)
iqr = grouper.quantile([0.75, 0.25]).groupby("Continent_Code").agg(np.subtract.reduce)
iqr

internet_users_per_100       Math    Reading    Science
Continent_Code                                                         
AF                            5.159918    3.60590    5.59810    5.32915
AS                           34.648353  127.46290  100.58830  105.66620
EU                           18.846156   39.68190   47.30030   41.16200
OC                            1.831185    0.66355    3.18505    1.65480
SA                            9.205070   29.51545   15.79580   25.58915
NaN                          17.123159   69.37510   73.65860   80.53250

In [17]:
df.groupby("Continent_Code", dropna=False).count()

Country Code  internet_users_per_100  Math  Reading  Science
Continent_Code                                                              
AF                         2                       2     2        2        2
AS                        17                      17    17       17       17
EU                        37                      37    37       37       37
OC                         2                       2     2        2        2
SA                         7                       7     7        7        7
NaN                        5                       5     5        5        5

2. Determine whether there is any difference (on the average) for the math scores among European (EU) and Asian (AS) countries (assume normality and equal variances). Draw side-by-side box plots.

In [98]:
import seaborn as sns

In [102]:
avg_math = df.groupby(["Continent_Code", "Country Code"])[["Math"]].mean()
avg_math

Math
Continent_Code Country Code          
AF             DZA           359.6062
               TUN           366.8180
AS             ARE           427.4827
               CHN           531.2961
               GEO           403.8332
...                               ...
SA             CHL           422.6714
               COL           389.6438
               PER           386.5606
               TTO           417.2434
               URY           417.9919

[65 rows x 1 columns]

In [273]:
avg_math.loc["AS"].mean()

Math    466.216647
dtype: float64

In [274]:
avg_math.loc["EU"].mean()

Math    477.981449
dtype: float64

In [275]:
stats.ttest_ind(avg_math.loc["EU"], avg_math.loc["AS"])

Ttest_indResult(statistic=array([0.87005532]), pvalue=array([0.38826888]))

The low t value indicates that there is no significant difference between European and Asian countries for mean mathematics scores.

# EXERCISE 4. The sample dataset has placement test scores (out of 100 points) for four subject areas: English, Reading, Math, and Writing. Students in the sample completed all 4 placement tests when they enrolled in the university. Suppose we are particularly interested in the English and Math sections, and want to determine whether students tended to score higher on their English or Math test, on average. We could use a paired t test to test if there was a significant difference in the average of the two tests.

In [30]:
grade = pd.read_csv("/content/students_2014 - students_2014.csv")
grade.head()

ids        bday enrolldate  ... CommuteTime  SleepTime StudyTime
0  43783   3/22/1995        NaN  ...         NaN        7.0       1.0
1  20278    1/1/1995        NaN  ...         NaN        5.0       2.0
2  20389  12/31/1994        NaN  ...         NaN        8.0       7.0
3  22820   12/1/1994        NaN  ...         NaN        2.0       6.0
4  24559  11/10/1994        NaN  ...         NaN        7.0       3.0

[5 rows x 23 columns]

In [39]:
grade_non_null = grade.loc[:, ["English", "Math"]].dropna()

In [40]:
grade_non_null.shape

(398, 2)

In [49]:
grade_non_null[["English", "Math"]].corr()

English      Math
English  1.000000  0.243014
Math     0.243014  1.000000

In [51]:
stats.ttest_rel(grade_non_null["English"], grade_non_null["Math"])

Ttest_relResult(statistic=36.312568981719856, pvalue=3.0710987192210606e-128)

In [52]:
grade_non_null.mean()

English    82.744095
Math       65.446834
dtype: float64

In [53]:
grade_non_null["English"].mean() - grade_non_null["Math"].mean()

17.297261306532633

-English and Math scores were weakly and positively correlated (r = 0.243, p < 0.001).

-There was a significant average difference between English and Math scores (t = 36.313, p < 0.001).

-On average, English scores were 17.30 points higher than Math scores